presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadataFactory.java

```diff
@@ -145,9 +151,12 @@ public HiveMetadataFactory(
 
     public HiveMetadata create()
     {
+        // Create separated CachingHiveMetastore for each SemiTransactionalHiveMetastore
+        CachingHiveMetastore perTransactionCache = new CachingHiveMetastore(metastore, cachingService);
```

Comment:
Inline, and break `new SemiTransactionalHiveMetastore` up into one line per argument. A comment like `// per-txn cache` will probably be sufficient.